src/masternodes/rpc_poolpair.cpp

```diff
@@ -1026,6 +1038,195 @@ UniValue testpoolswap(const JSONRPCRequest& request) {
+                                                         "\"amountFrom\":\"0.001\","
+                                                         "\"to\":\"Address\","
+                                                         "\"tokenTo\":\"Token2\","
+                                                         "\"maxPrice\":\"0.01\""
+                                                         "}' '[{\"txid\":\"id\",\"vout\":0}]'")
+                       + HelpExampleRpc("testcompositeswap", "'{\"from\":\"MyAddress\","
+                                                         "\"tokenFrom\":\"MyToken1\","
+                                                         "\"amountFrom\":\"0.001\","
+                                                         "\"to\":\"Address\","
+                                                         "\"tokenTo\":\"Token2\","
+                                                         "\"maxPrice\":\"0.01\""
+                                                         "}' '[{\"txid\":\"id\",\"vout\":0}]'")
+               },
+    }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload()) {
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot create transactions while still in Initial Block Download");
+    }
+    pwallet->BlockUntilSyncedToCurrentChain();
+
+    int targetHeight = chainHeight(*pwallet->chain().lock()) + 1;
+    if (targetHeight < Params().GetConsensus().FortCanningHeight) {
+        throw JSONRPCError(RPC_INVALID_REQUEST, "compositeswap is available post Fort Canning");
+    }
+
+    RPCTypeCheck(request.params, {UniValue::VOBJ, UniValue::VARR}, true);
+
+    CPoolSwapMessageV2 poolSwapMsgV2{};
+    CPoolSwapMessage& poolSwapMsg = poolSwapMsgV2.swapInfo;
+    CheckAndFillPoolSwapMessage(request, poolSwapMsg);
+
+    // test execution and returns execution result
+    Res res = Res::Ok();
+    {
+        LOCK(cs_main);
+        CCustomCSView dummy(*pcustomcsview); // create dummy cache for test state writing
+        // If no direct swap found search for composite swap
+        if (!dummy.GetPoolPair(poolSwapMsg.idTokenFrom, poolSwapMsg.idTokenTo)) {
+
+            auto compositeSwap = CPoolSwap(poolSwapMsg, targetHeight);
+            res = compositeSwap.TextExecuteSwap(dummy);
+
+            if (!res.ok) {
+                std::string errorMsg{"Cannot find usable pool pair."};
+                if (!compositeSwap.errors.empty()) {
+                    errorMsg += " Details: (";
+                    for (size_t i{0}; i < compositeSwap.errors.size(); ++i) {
+                        errorMsg += '"' + compositeSwap.errors[i].first + "\":\"" +  compositeSwap.errors[i].second + '"' + (i + 1 < compositeSwap.errors.size() ? "," : "");
+                    }
+                    errorMsg += ')';
+                }
+                throw JSONRPCError(RPC_INVALID_REQUEST, errorMsg);
+            }
+        }
+
+    }
+
+    return UniValue(res.msg);
```

Comment:
No need to wrap in UniValue, just return res.msg.

Reply:
Okay